test/framework/src/main/java/org/elasticsearch/test/TransportVersionUtils.java

```diff
@@ -24,7 +24,8 @@
 public class TransportVersionUtils {
 
     private static final NavigableSet<TransportVersion> RELEASED_VERSIONS = Collections.unmodifiableNavigableSet(
-        new TreeSet<>(TransportVersion.getAllVersions())
+        // Exclude patch versions since they break the semantics of methods like `randomVersionBetween`
+        TransportVersion.getAllVersions().stream().filter(v -> v.isPatchVersion() == false).collect(Collectors.toCollection(TreeSet::new))
```

Comment:
I wonder if this should create a new version where the patch version in the linked-list is always `null` as well, so supports won't check any patch versions unintentionally in tests.